apps/publish_manifest_lists.py

```diff
@@ -0,0 +1,53 @@
+import json
+import os
+from os.path import expanduser
+
+from helpers import cmd, jobserv_get, status
+from tag_manager import TagMgr
+
+
+def publish_manifest_lists(project: str = "", build_num: str = "", ota_lite_tag: str = ""):
+    if not project:
+        project = os.environ["H_PROJECT"]
+    if not build_num:
+        build_num = os.environ["H_BUILD"]
+    if not ota_lite_tag:
+        ota_lite_tag = os.environ["OTA_LITE_TAG"]
+    factory, _ = project.split("/")
+    latest_tag = TagMgr(ota_lite_tag).tags[0][0]
+
+    status("Publish manifest lists for containers")
+    build = jobserv_get(f"/projects/{project}/builds/{build_num}/")["data"]["build"]
+
+    tags = {}
+
+    manifests_dir = expanduser("~/.docker/manifests")
+    os.makedirs(manifests_dir, exist_ok=True)
+
+    for run in build["runs"]:
+        status(f" Looking for containers built by {run['name']}")
+        run = jobserv_get(run["url"])["data"]["run"]
+        needle = run["url"] + "manifests/"
+        for a in run.get("artifacts"):
+            if a.startswith(needle):
+                _, container_name, _ = a.rsplit("/", 2)
+                tags[container_name] = 1
+                mf = jobserv_get(a)
+                path = os.path.join(manifests_dir, a[len(needle):])
+                try:
+                    os.mkdir(os.path.dirname(path))
+                except FileExistsError:
+                    pass
+                with open(path, "w") as f:
+                    json.dump(mf, f)
+
+    for tag in tags.keys():
+        mfdir = os.path.join(manifests_dir, tag)
+        names = os.listdir(mfdir)
+        tag = tag.replace(f"_{factory}_", f"/{factory}/")
+        if mfdir.endswith(latest_tag):
+            tag = tag.replace(f"-{latest_tag}", f":{latest_tag}")
```

Comment:
It would be great to add a comment containing a tag format this logic composes.

Reply:
https://github.com/foundriesio/ci-scripts/pull/198/commits/7ca2bc845b23d95f89cd604c5c3022b5e2bad22c